testing/sdk_install.py

```diff
@@ -176,41 +175,29 @@ def uninstall(
         zk=None):
     start = time.time()
 
-    global _installed_service_names
-    try:
-        _installed_service_names.remove(service_name)
-    except KeyError:
-        pass  # allow tests to 'uninstall' up-front
-
     log.info('Uninstalling {}'.format(service_name))
 
     try:
         retried_uninstall_package_and_wait(package_name, service_name=service_name)
     except Exception:
-        log.info('Got exception when uninstalling {}'.format(service_name))
-        log.info(traceback.format_exc())
+        log.exception('Got exception when uninstalling {}'.format(service_name))
         raise
-    finally:
-        log.info('Reserved resources post uninstall:')
-        sdk_utils.list_reserved_resources()
 
     cleanup_start = time.time()
 
     try:
         if sdk_utils.dcos_version_less_than('1.10'):
+            # 1.9 and earlier: Run janitor to unreserve resources
             log.info('Janitoring {}'.format(service_name))
             retried_run_janitor(service_name, role, service_account, zk)
         else:
+            # 1.10 and later: Wait for uninstall scheduler to finish and be removed by Cosmos
             log.info('Waiting for Marathon app to be removed {}'.format(service_name))
             sdk_marathon.retried_wait_for_deployment_and_app_removal(
                 sdk_marathon.get_app_id(service_name), timeout=TIMEOUT_SECONDS)
     except Exception:
-        log.info('Got exception when cleaning up {}'.format(service_name))
-        log.info(traceback.format_exc())
+        log.exception('Got exception when cleaning up {}'.format(service_name))
         raise
```

Comment:
Same comment as above? Also, the removal of the `finally` blocks in both cases changes behaviour subtly in that the resource list is *not* printed if there was an exception.

Reply:
The previous `list_reserved_resources()` functionality was moved to the following check, where it now throws if there are leftover resources (implies an uninstall bug).

If an exception is thrown here, there isn't much value in checking for leftover resources -- it's expected that there would be some if the uninstall trigger itself failed.